audioread/ffdec.py

```diff
@@ -73,6 +75,18 @@ def run(self):
                 # Stream closed (EOF).
                 break
 
+class WriterThread(threading.Thread):
+    """A thread that writes data to a filehandle
+    """
+    def __init__(self, fh, audio=None):
+        super(WriterThread, self).__init__()
+        self.fh = fh
+        self.audio = audio
+        self.daemon = True
+
+    def run(self):
+        self.fh.write(self.audio.read())
```

Comment:
This currently reads the entire input stream into memory and then writes into FFmpeg. We should do this in a more streaming way by reading a block from `self.audio` and writing it to `self.fh`, one block at a time, in a loop.